### DISCUSSION TITLE
torch_geometric.nn.models.VGAE has no provision for batch processing, right?

### DISCUSSION THREAD

**Discussion by survivebycoding** [pyg-team/pytorch_geometric#9814] [Q&A]
This function has no provision for batch processing right?

https://pytorch-geometric.readthedocs.io/en/2.5.2/generated/torch_geometric.nn.models.VGAE.html#torch_geometric.nn.models.VGAE

If there is a way, can someone kindly help me out with it please?

**Comment by Romain-Nicolle** [comment]
Hello,

I am not an expert but I will try to help you.
So, the source code for the VGAE model is [there](https://github.com/pyg-team/pytorch_geometric/blob/master/torch_geometric/nn/models/autoencoder.py).

[Here](https://github.com/pyg-team/pytorch_geometric/blob/master/benchmark/kernel/graph_sage.py) you can find the source code for the GraphSAGE model which support batch processing. The message_passing_convolution used by the GraphSAGE model is the SAGEConv, whose source code is [there](https://github.com/pyg-team/pytorch_geometric/blob/master/torch_geometric/nn/conv/sage_conv.py).
Interestingly, even though the SAGEConv doesn't seem to support batch_processing, the GraphSAGE model can support it, because it is written that way in the forward function.

So I gues, you could do that too. Rewrite some parts of the code in order for the functions to handle batches ?

Besides, have you checkes this [question](https://github.com/pyg-team/pytorch_geometric/discussions/3046) related to batches and VGAE ? Maybe it could help you ? Depending on whether you want to have mini-batches of one big graph or batches of several small graphes.

Hope that helps.

Best,
